sql/core/src/test/scala/org/apache/spark/sql/execution/SparkPlanSuite.scala

```diff
@@ -143,6 +143,48 @@ class SparkPlanSuite extends QueryTest with SharedSparkSession {
       }
     }
   }
+
+  test("SPARK-39854: replaceWithAliases should keep the order of Generate children") {
```

Comment:
Thank you for providing the end-to-end test. Can we have a test case in `NestedColumnAliasingSuite.scala` because we are touching `NestedColumnAliasing`?

Reply:
Hi @dongjoon-hyun , thanks for taking time looking into this!

The test is put in the `core` module (instead of the `catalyst` module) because it has dependency on things like `SparkSession`, `sql.functions`, which is only available there -- not sure where is the best place to put it, or are there better ways to re-write the test 😓 .

Reply:
If we add an end-to-end test, the Apache Spark test time increases prohibitively. We prefer to narrow down the issue and have isolated unit tests. So, in this PR, `NestedColumnAliasingSuite.scala` is the best place to have a test coverage. In short, I don't think we need a heavy end-to-end test case like this. We need to have a test case similar to the other in `NestedColumnAliasingSuite.scala`.

Reply:
I understand. will try to create a test case in `NestedColumnAliasingSuite.scala` instead.